### DISCUSSION TITLE
Dynamically changing model details as it runs

### DISCUSSION THREAD

**Discussion by Dknife7** [openTCS/opentcs#70] [Q&A] [answered]
Firstly, thank you so much for creating an wonderful project and make it open source. It has been really helpful for our project.

So, as in our project, we need to route the vehicle differently according to the status of the vehicle as well as the status of the location. eg, when it is loaded, it cannot use some of the path in routing. After going through the documents, I found one good solution could be that set the loaded vehicle to a routinggroup, and give every path a explicit Fowardcost<loaded> value. In such a way, when the vehicle is assigned to the loaded group, it will use the explicit value to calculate route, when it's not assigned it will use the default routing strategy. In the meantime, we also need to adjust the path property on the go. eg. when certain path is occupied by stock, it is allowed for unloaded vehicle to pass but not the loaded vehicle, then we need to clean this path's Fowardcost<loaded> property so that the loaded routinggroup will not use this path for routing. When the stock is removed, then we need to reassign this path with a Fowardcost<loaded> property, and then the loaded vehicles can use this path.
I see that changing the vehicle properties is kind easy through the Web APIs as in the docs.
My concern will be is it possible in OpenTCS to modify the path properties as the kernel is running？If yes, which direction of the code should I dive into. any hint will be really appreciated!

**Comment by swltr** [comment, marked as answer]
With the Java API (not the web API), you can use `TCSObjectService.updateObjectProperty()` to modify properties of any object, including `Path`s. Do note, though, that the API does not specify whether the changes you make there are persisted or not, so restarting the kernel may or may not clear all that information.
